### DISCUSSION TITLE
Iinteresting to see a similar project

### DISCUSSION THREAD

**Discussion by Declow** [paveldedik/ludic#66] [General]
Hey

Just came across your project. 6 or so months ago I started to develop my own framework using HTMX and python.

I have not yet had to time to really deep dive into all of your code but some of it has given me some inspiration on how to approach some things. I hope you might get the same benefit by seeing a different approach to tackle this problem.

I really like what I'm seeing and keep up the good work!

If you are interested in a similar project take a look at https://github.com/Declow/uiwiz/

**Comment by paveldedik** [comment]
Hi @Declow 

thank you very much for letting me know about your project, I'll definitely have a proper look. And at first, I was also considering using `with` statements to nest components.

**Comment by hasansezertasan** [comment]
Great work 🚀🚀🚀, happy to see HTMX adoption is spreading.

> And at first, I was also considering using `with` statements to nest components.

If you decide to use `with` statement, please just add it as an option and don't make it mandatory 😇.

**Reply by paveldedik** [reply]
No worries, Ludic is taking different direction :)

**Comment by Declow** [comment]
Well when I was starting I had seen quite a few projects that had already done the nesting using classes.
I kinda prefer the with statements over nesting using class instantiation. My reasoning not going with nesting class initialization was when editing ui I wanted to be able to quickly move the indentation of a single element. 
But it also creates a bit of complexity when the instance is created.

@hasansezertasan I was not trying to sell context manager approach vs class initialization 😅

**Reply by hasansezertasan** [reply]
> @hasansezertasan I was not trying to sell context manager approach vs class initialization 😅

Well, you can not sell it to me 😂. Joke aside, I was just expressing my taughts about context manager take, it wasn't a reply to your approach. I respect that.

One more thing, I also don't like using lowercase names for classes that represents HTML tags... Which I still respect but don't enjoy much, still it's the devs choice 🤓.
